File: python/automation_agent/agent/setup/session.py
"""ADK session-service factory for the configured backend.

The session service holds the durable suspend/resume history that lets a parked fix run
continue after a process restart. Only the long-running fix loop needs durability;
ephemeral one-shot runners (explore/analyze/triage) keep using an in-memory session.

    memory    -> in-process; tests and ephemeral local runs (today's behavior, default)
    sqlite    -> file-backed via adk SqliteSessionService; durable local runs (later phase)
    firestore -> cloud, adk's native google.adk.integrations.firestore service (later phase)

Keeping the backend imports here respects the ARCH boundary (infrastructure SDKs live
under ``agent.setup``).
"""

from __future__ import annotations

from typing import TYPE_CHECKING

from google.adk.sessions import BaseSessionService, InMemorySessionService

if TYPE_CHECKING:
    from automation_agent.config import Config


def new_session_service(cfg: Config) -> BaseSessionService:
    """Build the ADK session service for the configured backend. sqlite and firestore land
    in later phases (firestore uses adk's native FirestoreSessionService, not a hand-roll —
    only the park store is custom)."""
    from automation_agent.config import SessionBackend

    if cfg.session_backend == SessionBackend.MEMORY:
        return InMemorySessionService()
    raise NotImplementedError(
        f"session backend {cfg.session_backend!r} not yet implemented "
        "(sqlite/firestore land in a later phase); use SESSION_BACKEND=memory"
    )

File: kotlin/src/main/kotlin/com/automation/agent/agent/setup/Session.kt
/*
 * The ADK session-service factory: builds the durable suspend/resume session backend for the
 * configured SESSION_BACKEND. The session service holds the history that lets a parked fix run
 * continue after a restart; only the long-running fix loop needs it (ephemeral one-shot runners
 * keep their own in-memory session).
 *
 *   memory    -> in-process (today's behavior, default)
 *   sqlite    -> file-backed via the hand-rolled [SqliteSessionService]; durable local runs
 *   firestore -> cloud, via the hand-rolled Firestore session service; durable cloud runs
 */
package com.automation.agent.agent.setup

import com.google.adk.kt.sessions.InMemorySessionService
import com.google.adk.kt.sessions.SessionService
import com.automation.agent.config.Config
import com.automation.agent.config.SessionBackend
import java.io.File

/** Build the ADK session service for the configured session backend. */
fun newSessionService(cfg: Config): SessionService =
    when (cfg.sessionBackend) {
        SessionBackend.MEMORY -> InMemorySessionService()
        // Absolute path so the session service and the park store open the very same file
        // regardless of the process working directory.
        SessionBackend.SQLITE -> SqliteSessionService(File(cfg.sqliteDsn).absolutePath)
        SessionBackend.FIRESTORE -> FirestoreSessionService(cfg.firestoreProject, "${cfg.firestoreCollection}_sessions")
    }
